rfc/cart-api-v1.md

```diff
@@ -0,0 +1,179 @@
+# Version 1
+
+| Issue | Status |
+| ----- | ------ |
+| #88   | Draft  |
+
+## Data Models
+
+There are two data models required for Cart API: `Cart` and `CartItem`. They are
+designed in the way that allows multiple types of entity in a single `Cart`.
+
+### `Cart`
+| Property | Type | Description | Note |
+| --- | --- | --- | --- |
+| `id` | UUID | Cart ID, auto-generated by the web API | - |
+| `user_id` | UUID | The ID of the user that creates the collection | - |
+| `cart_name` | `str` | Cart Name: If it is `NULL`, the cart is the default one for a corresponding user. | **NEED MORE DISCUSSION** |
+| `default_cart` | `bool` | Flag to indicate whether this cart is the default one for a corresponding user (`False` by default) | **NEED MORE DISCUSSION** |
+
+For the details on the default cart, see `GET /resources/carts/{id}`.
+
+### `CartItem`
+| Property | Type | Description |
+| --- | --- | --- |
+| `id` | `str` | Cart Item ID, hashed by the combination of `` |
+| `cart_id` | UUID | Associated `Cart.id` |
+| `entity_id` | UUID | Entity ID |
+| `entity_type` | `str` | Entity Type |
+| `bundle_uuid` | UUID | Bundle ID |
```

Comment:
How should we handle entities with more than one bundle (i.e. non-files)

Reply:
If entity can have one or many bundles,

``` Entity --(1)----(1..n)--> Bundle ```

then, each cart item is per entity per bundle per bundle version.

Reply:
That's a lot of cart items.  
I forgot that we can also store lists/sets which may work depending on the use case of the bundle information.  I don't actually remember why we decided to store the bundle here.  I don't think DSS collections use it.